site/src/helpers/substitutions.ts

```diff
@@ -0,0 +1,32 @@
+export const labelSubstitutions: Record<string, string> = {
+  Div: 'Division',
+  "Int'l": 'International',
+  US: 'United States',
+};
+
+export function expandAbbreviation(label: string): string {
+  let expandedLabel = label;
+
+  for (const [abbrev, substitution] of Object.entries(labelSubstitutions)) {
+    const regex = new RegExp(`\\b${abbrev}\\b`, 'gi');
+    expandedLabel = expandedLabel.replace(regex, substitution);
+  }
+
+  return expandedLabel;
+}
+
+export function findCommonLabelPrefix(labels: string[]): string {
+  if (labels.length === 0) return '';
+
+  const expandedLabels = labels.map(expandAbbreviation); // expand any words in labels that may be abbreviated
+
+  expandedLabels.sort();
+  const first = expandedLabels[0].split(' ');
+  const last = expandedLabels[expandedLabels.length - 1].split(' ').filter((word) => !word.startsWith('(')); // ignore any words within parentheses
```

Comment:
I guess this is one way of terminating match since "I. Upper-Division Writing (min grade C-) (subject)" will be different from "I. Upper-Division Writing grade C-) (subject 2)", but I think it's probably more understandable if we just remove all text after the first open paren (`.replace(/\s\(.*/g, '')`) before filtering like this.

Also you can get last item with `arr.at(-1)` instead of `arr[arr.length - 1]`